### PR TITLE
RM-8463 - bugfix/RM-8463-BocAutoCompleteReferenceValue-Tab-Key-Is-Broken

### PR DESCRIPTION
**Pull request by MichaelKetting**
https://re-motion.atlassian.net/browse/RM-8463

### CODE REVIEW

File: Remotion/ObjectBinding/Web.ClientScript/Scripts/BocAutoCompleteReferenceValue.UI.ts [hunk 1]
```diff
@@ -400,9 +398,13 @@ namespace Remotion.BocAutoCompleteReferenceValue
...
-                            // stop default to prevent a form submit, Opera needs special handling
-                            blockSubmit = true;
+                        if (options.isAutoPostBackEnabled) {
+                            // stop default on auto-postback since the change-event already takes care of this.
+                            event.preventDefault();
+                            return false;
+                        } else if (wasVisible) {
+                            // stop default for visible dropdown options since RETURN should simply select the current option without triggering a postback.
+                            event.preventDefault();
                             return false;
```
Comment (MichaelKetting) [inline Remotion/ObjectBinding/Web.ClientScript/Scripts/BocAutoCompleteReferenceValue.UI.ts:408]: Consider line 415 (ESC) also `event.preventDefault()` for `return false;`

**Review by a-ctor — APPROVED** [review]

File: Remotion/ObjectBinding/Web.ClientScript/Scripts/BocAutoCompleteReferenceValue.UI.ts (hunk 1 above)
Comment (MichaelKetting) [reply Remotion/ObjectBinding/Web.ClientScript/Scripts/BocAutoCompleteReferenceValue.UI.ts]: @a-ctor this discussion about preventDefault wasn't considered during the review. The latest code now uses the updated preventDefault semantsics.

**Review by a-ctor — APPROVED** [review]

File: Remotion/ObjectBinding/Web.ClientScript/Scripts/BocAutoCompleteReferenceValue.UI.ts (hunk 1 above)
Comment (a-ctor) [reply Remotion/ObjectBinding/Web.ClientScript/Scripts/BocAutoCompleteReferenceValue.UI.ts]: I don't know if we need `.preventDefault` here but it does not hurt. I don't think that the browser has a default behavior for ESC

Comment (MichaelKetting) [reply Remotion/ObjectBinding/Web.ClientScript/Scripts/BocAutoCompleteReferenceValue.UI.ts]: This is true, but given the original code defined it as `false`, the `preventDefault()` preserves the original semantics regardless of it being needed.

**Pull request merged**